ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -77,7 +243,96 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
           done();
         });
       };
+      var vocabularyComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <VocabularyComponent
+            vocabulary={vocabulary}
+            terms={vocabulary.terms}
+            reportError={reportError}
+            addTerm={addTerm}
+            repoSlug="repo"
+            ref={afterMount}
+          />
+        );
+      assert.equal(
+        vocabularyComponentRendered.state.terms.length,
+         2
+      );
+      // check term title
+      var listOfLinks = React.addons.TestUtils.
+        scryRenderedDOMComponentsWithTag(
+          vocabularyComponentRendered,
+          'a'
+      );
+      assert.equal(listOfLinks.length, 2);
+      var linkHeader = listOfLinks[0].getDOMNode();
+      assert.equal(linkHeader.innerHTML, vocabulary.name);
 
+      //test items
+      var devItems = React.addons.TestUtils.
+        findRenderedDOMComponentWithClass(
+          vocabularyComponentRendered,
+          'panel-body'
+      );
+      var itemList = React.addons.TestUtils.
+        scryRenderedDOMComponentsWithTag(
+          devItems,
+          'li'
+      );
+      assert.equal(itemList.length, 2);
+      //test enter text in input text
+      var inputNode = React.addons.TestUtils.
+        findRenderedDOMComponentWithTag(
+          vocabularyComponentRendered,
+          'input'
+      );
+      React.addons.TestUtils.Simulate.change(
+        inputNode,
+        {target: {value: 'test12'}}
+      );
+      assert.equal(
+        'test12',
+        vocabularyComponentRendered.state.newTermLabel
+      );
+    }
+  );
+  QUnit.test('Assert that ajax fail VocabularyComponent',
+    function(assert) {
+      var done = assert.async();
+      var vocabulary = {
+        "id": 2,
+        "slug": "easy",
+        "name": "easy",
+        "description": "easy",
+        "vocabulary_type": "m",
+        "required": false,
+        "weight": 2147483647,
+        "terms": []
+      };
+      var addTermCalled = 0;
+      var errorMessage;
+      var addTerm = function() {
+        addTermCalled += 1;
+      };
+      var reportError = function(msg) {
+        errorMessage = msg;
+      };
+      var afterMount = function(component) {
+        // wait for calls to populate form
+        var node = React.findDOMNode(component);
+        var textbox = $(node).find("input")[0];
+        component.state.newTermLabel = "term2";
```

Comment:
Use `component.setState` instead. `state` is not supposed to be mutated like this